code/game/atoms_movable.dm

```diff
@@ -639,31 +640,49 @@
 
 /atom/movable/Exited(atom/movable/gone, direction)
 	. = ..()
-	if(gone.area_sensitive_contents)
-		for(var/atom/movable/location as anything in get_nested_locs(src) + src)
-			LAZYREMOVE(location.area_sensitive_contents, gone.area_sensitive_contents)
+	if(gone.important_recursive_contents) //with a loop for every key in arrived's important_recursive_contents
+		for(var/channel in gone.important_recursive_contents)
+			for(var/atom/movable/location as anything in get_nested_locs(src) + src)
+				LAZYREMOVEASSOC(location.important_recursive_contents, channel, gone.important_recursive_contents[channel])
```

Comment:
for those wondering what i meant by "semi-automatic compile time loop unrolling", i mean this 
![Screenshot_419](https://user-images.githubusercontent.com/15794172/125595047-95741570-5c1d-4a45-8282-f1e067abeb45.png)